dask/base.py

```diff
@@ -139,11 +141,20 @@ def normalize_function(func):
 
 
 normalize_token = Dispatch()
-normalize_token.register((int, float, str, tuple, list), lambda a: a)
-normalize_token.register(object,
-        lambda a: normalize_function(a) if callable(a) else a)
+normalize_token.register((int, float, str, unicode, bytes, tuple, list,
+                         type(None)),
+                         identity)
 normalize_token.register(dict, lambda a: tuple(sorted(a.items())))
```

Comment:
Consider OrderedDict vs dict? These may sometimes result in different results but will have the same token...